File: pueblo-golf-tournament-api/Dto/PlayerTeeTimeScheduleDto.cs
using pueblo_golf_tournament_api.Base;
using pueblo_golf_tournament_api.Utilities.Enums;

namespace pueblo_golf_tournament_api.Dto
{
    public class PlayerTeeTimeScheduleDto : BaseDto
    {
        public long PlayerId { get; set; }
        public long TeeTimeScheduleId { get; set; }
        public HoleTypeEnum HoleType { get; set; }
    }
}

File: pueblo_golf_tournament_mobile/lib/dto/model/player-tee-time-schedule-dto.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pueblo_golf_tournament_mobile/base/base-dto.dart';

part 'player-tee-time-schedule-dto.g.dart';

@JsonSerializable()
class PlayerTeeTimeScheduleDto extends BaseDto {
  final int playerId;
  final int teeTimeScheduleId;
  final int holeType;

  PlayerTeeTimeScheduleDto(
      {required super.id,
      required this.playerId,
      required this.teeTimeScheduleId,
      required this.holeType});

  factory PlayerTeeTimeScheduleDto.fromJson(Map<String, dynamic> json) =>
      _$PlayerTeeTimeScheduleDtoFromJson(json);
  Map<String, dynamic> toJson() => _$PlayerTeeTimeScheduleDtoToJson(this);
}
